index.js

```diff
@@ -1,81 +1,143 @@
-}
+    homebridge.registerAccessory("homebridge-mi-led-desk-lamp", "mi-led-desk-lamp", MiLedDesklamp);
+};
 
 class MiLedDesklamp {
 
     constructor(log, config) {
         // Setup configuration
-        this.log = log
-        this.name = config['name'] || 'Mi desk lamp'
+        this.log = log;
+        this.name = config['name'] || 'Mi desk lamp';
         if (!config['ip']) {
-            this.log('No IP address define for', this.name)
-            return
+            this.log('No IP address define for', this.name);
+            return;
         }
         if (!config['token']) {
-            this.log('No token define for', this.name)
-            return
+            this.log('No token define for', this.name);
+            return;
         }
-        this.ip = config['ip']
-        this.token = config['token']
+        this.ip = config['ip'];
+        this.token = config['token'];
 
         // Setup services
-        this.switch = new Service.Switch(this.name)
-        this.switch.getCharacteristic(Characteristic.On)
+        this.lamp = new Service.Lightbulb(this.name);
+        this.lamp.getCharacteristic(Characteristic.On)
             .on('get', this.getState.bind(this))
-            .on('set', this.setState.bind(this))
+            .on('set', this.setState.bind(this));
+
+        this.lamp.getCharacteristic(Characteristic.Brightness)
+            .on('get', this.getBrightness.bind(this))
+            .on('set', this.setBrightness.bind(this));
 
-        this.listenLampState().catch(error => this.log.error(error))
+        this.lamp.getCharacteristic(Characteristic.ColorTemperature)
+            .on('get', this.getColorTemperature.bind(this))
+            .on('set', this.setColorTemperature.bind(this));
+
+        this.listenLampState().catch(error => this.log.error(error));
     }
 
     async getLamp() {
-        if (this.lampDevice) return this.lampDevice
-        this.log('Connect to device')
+        if (this.lampDevice) return this.lampDevice;
+        this.log('Connect to device');
         try {
-            this.lampDevice = await miio.device({address: this.ip, token: this.token})
+            this.lampDevice = await miio.device({address: this.ip, token: this.token});
         } catch (e) {
-            this.lampDevice = undefined
-            this.log.error('Device not connected', e)
+            this.lampDevice = undefined;
+            this.log.error('Device not connected', e);
         }
-        return this.lampDevice
+        return this.lampDevice;
     }
 
     async listenLampState(){
-        const device = await this.getLamp()
-        device.on('powerChanged', isOn => this.switch.getCharacteristic(Characteristic.On).updateValue(isOn))
+        const device = await this.getLamp();
+        device.on('powerChanged', isOn => this.lamp.getCharacteristic(Characteristic.On).updateValue(isOn));
+	    device.on('colorChanged', color => this.lamp.getCharacteristic(Characteristic.ColorTemperature).updateValue(color));
```

Comment:
Brightness state is missing, is it normal? or miio library can not manage it?

Reply:
Added.